File: packages/python/src/trend_buddy_shared/contracts/commands/backtest.py
from dataclasses import dataclass, field
from datetime import datetime
from decimal import Decimal
from typing import Any

from trend_buddy_shared.contracts.domain.instrument import Timeframe


@dataclass
class BacktestRequest:
    strategy_id: str
    instrument_id: str
    timeframe: Timeframe
    start_time: datetime
    end_time: datetime
    initial_capital: Decimal
    fees: Decimal = Decimal("0")
    slippage: Decimal = Decimal("0")
    parameters: dict[str, Any] = field(default_factory=dict)

File: packages/go/contracts/commands/backtest.go
package commands

import (
	"time"

	"trend-buddy/packages/go/contracts/domain"
)

type BacktestRequest struct {
	StrategyID     domain.StrategyID   `json:"strategy_id"`
	InstrumentID   domain.InstrumentID `json:"instrument_id"`
	Timeframe      domain.Timeframe    `json:"timeframe"`
	StartTime      time.Time           `json:"start_time"`
	EndTime        time.Time           `json:"end_time"`
	InitialCapital domain.Decimal      `json:"initial_capital"`
	Fees           domain.Decimal      `json:"fees,omitempty"`
	Slippage       domain.Decimal      `json:"slippage,omitempty"`
	Parameters     map[string]any      `json:"parameters,omitempty"`
}
